liquibase.exception.DatabaseException: class path resource [config/liquibase/data/blob/hipster.txt] cannot be resolved to URL because it does not exist

##### **Overview of the issue**

My JHipster app fails to start after importing JDL:

```
entity Blog {
  name String required minlength(3),
  handle String required minlength(2)
}

entity Entry {
  title String required,
  content TextBlob required,
  date Instant required
}

entity Tag {
  name String required minlength(2)
}

relationship ManyToOne {
  Blog{user(login)} to User,
  Entry{blog(name)} to Blog
}

relationship ManyToMany {
  Entry{tag(name)} to Tag{entry}
}

paginate Entry, Tag with infinite-scroll
```

```----------------------------------------------------------
	Application 'blog' is running! Access URLs:
	Local: 		http://localhost:8080/
	External: 	http://192.168.50.28:8080/
	Profile(s): 	[dev, swagger]
----------------------------------------------------------
2019-06-22 13:40:32.675 ERROR 5647 --- [    blog-task-1] liquibase.changelog.ChangeSet            : Change Set config/liquibase/changelog/20190622193652_added_entity_Entry.xml::20190622193652-1-data::jhipster failed.  Error: class path resource [config/liquibase/data/blob/hipster.txt] cannot be resolved to URL because it does not exist
2019-06-22 13:40:32.679 ERROR 5647 --- [    blog-task-1] i.g.j.c.liquibase.AsyncSpringLiquibase   : Liquibase could not start correctly, your database is NOT ready: Migration failed for change set config/liquibase/changelog/20190622193652_added_entity_Entry.xml::20190622193652-1-data::jhipster:
     Reason: liquibase.exception.DatabaseException: class path resource [config/liquibase/data/blob/hipster.txt] cannot be resolved to URL because it does not exist

liquibase.exception.MigrationFailedException: Migration failed for change set config/liquibase/changelog/20190622193652_added_entity_Entry.xml::20190622193652-1-data::jhipster:
     Reason: liquibase.exception.DatabaseException: class path resource [config/liquibase/data/blob/hipster.txt] cannot be resolved to URL because it does not exist
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:637)
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:53)
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:83)
	at liquibase.Liquibase.update(Liquibase.java:202)
	at liquibase.Liquibase.update(Liquibase.java:179)
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:353)
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:305)
	at io.github.jhipster.config.liquibase.AsyncSpringLiquibase.initDb(AsyncSpringLiquibase.java:119)
	at io.github.jhipster.config.liquibase.AsyncSpringLiquibase.lambda$afterPropertiesSet$0(AsyncSpringLiquibase.java:94)
	at io.github.jhipster.async.ExceptionHandlingAsyncTaskExecutor.lambda$createWrappedRunnable$1(ExceptionHandlingAsyncTaskExecutor.java:78)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: liquibase.exception.DatabaseException: class path resource [config/liquibase/data/blob/hipster.txt] cannot be resolved to URL because it does not exist
```

##### **Motivation for or Use Case**

This JDL used to work before Faker.js was integrated.

##### **Reproduce the error**

Create a JHipster app, import the JDL above, restart your app.

##### **Suggest a Fix**

Not sure. Disabling Faker is a workaround.

##### **JHipster Version(s)**

```
blog@0.0.0 /Users/mraible/blog
└── generator-jhipster@6.1.1

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "org.jhipster.blog",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "6.1.1",
    "applicationType": "monolith",
    "baseName": "blog",
    "packageName": "org.jhipster.blog",
    "packageFolder": "org/jhipster/blog",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
    "useSass": true,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "testFrameworks": ["protractor"],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en", "es"]
  }
}

</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Blog {
  name String required minlength(3),
  handle String required minlength(2)
}
entity Entry {
  title String required,
  content TextBlob required,
  date Instant required
}
entity Tag {
  name String required minlength(2)
}
relationship ManyToOne {
  Blog{user(login)} to User,
  Entry{blog(name)} to Blog
}
relationship ManyToMany {
  Entry{tag(name)} to Tag{entry}
}

paginate Entry, Tag with infinite-scroll

</pre>
</details>


##### **Environment and Tools**

openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)

git version 2.20.1 (Apple Git-117)

node: v10.15.3

npm: 6.9.0

yeoman: 2.0.6

Docker version 18.09.2, build 6247962

docker-compose version 1.23.2, build 1110ad01

--- a/generators/entity-server/files.js
+++ b/generators/entity-server/files.js
@@ -83,7 +83,7 @@ const serverFiles = {
             templates: [{ file: 'config/liquibase/data/blob/hipster.png', method: 'copy', noEjs: true }]
         },
         {
-            condition: generator => generator.databaseType === 'sql' && generator.fieldsContainBlobOrImage === true,
+            condition: generator => generator.databaseType === 'sql' && generator.fieldsContainTextBlob === true,
             path: SERVER_MAIN_RES_DIR,
             templates: [{ file: 'config/liquibase/data/blob/hipster.txt', method: 'copy' }]
         },

--- a/generators/entity/index.js
+++ b/generators/entity/index.js
@@ -700,6 +700,7 @@ class EntityGenerator extends BaseBlueprintGenerator {
                 context.fieldsContainBigDecimal = false;
                 context.fieldsContainBlob = false;
                 context.fieldsContainImageBlob = false;
+                context.fieldsContainTextBlob = false;
                 context.fieldsContainBlobOrImage = false;
                 context.validation = false;
                 context.fieldsContainOwnerManyToMany = false;
@@ -845,6 +846,8 @@ class EntityGenerator extends BaseBlueprintGenerator {
                         }
                         if (field.fieldTypeBlobContent !== 'text') {
                             context.fieldsContainBlobOrImage = true;
+                        } else {
+                            context.fieldsContainTextBlob = true;
                         }
                     }
 

